docs/working-with-gatekeeper.md

```diff
@@ -0,0 +1,2802 @@
+# Use Gatekeeper to manage OPA policy in Karmada controller plane
+
+[Gatekeeper](https://github.com/open-policy-agent/gatekeeper) , is a customizable admission webhook for Kubernetes that enforces policies executed by the Open Policy Agent (OPA), a policy engine for Cloud Native environments hosted by [Cloud Native Computing Foundation ](https://cncf.io/).
+
+This document gives an example to demonstrate how to use the `Gatekeeper` to manage OPA policy. 
+
+##Start up Karmada clusters
+You just need to clone Karmada repo, and run the following script in Karmada directory. 
+
+```
+hack/local-up-karmada.sh
+```
+
+## Start Gatekeeper
+
+In this case, we will use Gatekeeper v3.7.2. Related deployment files are from [here](https://github.com/open-policy-agent/gatekeeper/blob/release-3.7/deploy/gatekeeper.yaml).
+
+1. Create resource objects of Gatekeeper in karmada controller plane, the content is as follows. 
+
+   ```console
+   kubectl config use-context karmada-apiserver
+   ```
+
+   ```yaml
+   apiVersion: v1
+   kind: Namespace
+   metadata:
+      labels:
+         admission.gatekeeper.sh/ignore: no-self-managing
+         control-plane: controller-manager
+         gatekeeper.sh/system: "yes"
+      name: gatekeeper-system
+   ---
+   apiVersion: apiextensions.k8s.io/v1
+   kind: CustomResourceDefinition
+   metadata:
+      annotations:
+         controller-gen.kubebuilder.io/version: v0.8.0
+      labels:
+         gatekeeper.sh/system: "yes"
+      name: assign.mutations.gatekeeper.sh
```

Comment:
This crd file seems so long, can paste the source link address or fold it?

Reply:
+1

Reply:
Remove crd files and add reference.